### PR TITLE
make lint: Flake8 tests to find syntax errors, etc.

### PR DESCRIPTION
**Pull request by cclauss**
Blocked by #1278 and #1279.

Add a __make lint__ command to run [flake8](http://flake8.pycqa.org) tests to find Python syntax errors and undefined names.  Also add a __make lint__ run to __.travis.yml__ and add __flake8__ to __test_requirements.txt__.

__E901,E999,F821,F822,F823__ are the "_showstopper_" flake8 issues that can halt the runtime with a SyntaxError, NameError, etc. Most other flake8 issues are merely "style violations" -- useful for readability but they do not effect runtime safety.
* ~F821: undefined name `name`~ TODO: Enable this once the others have been fixed.
* F822: undefined name `name` in `__all__`
* F823: local variable name referenced before assignment
* E901: SyntaxError or IndentationError
* E999: SyntaxError -- failed to compile a file into an Abstract Syntax Tree

### CODE REVIEW

**Comment by jdlrobson** [comment]
Awesome! I was wondering about this the other day!

**Comment by cclauss** [comment]
@jdlrobson Could I please request a review?  Scroll to the top of this page and click "Files changed" and then "Review changes".  The more "Approved" reviews that a PR has, the more confidence that maintainers will have in the usefulness of the PR.

**Comment by jdlrobson** [comment]
Will look later today!

File: openlibrary/catalog/amazon/amazon_to_arc.py [hunk 1]
```diff
@@ -24,9 +24,9 @@ def get(sock, host, url):
         if chunk_size == 0:
             break
         print(len(fp.read(chunk_size)))
-        print(repr(fp.read(2))
```
Comment (mekarpeles) [inline openlibrary/catalog/amazon/amazon_to_arc.py]: thank you :) this was flat out broken

**Comment by jdlrobson** [comment]
Also looking good to me

**Comment by mekarpeles** [comment]
Thanks @cclauss!

File: openlibrary/catalog/amazon/amazon_to_arc.py (hunk 1 above)
Comment (hornc) [reply openlibrary/catalog/amazon/amazon_to_arc.py]: I'm not sure /catalog/amazon is used by the main site in any way. There seems to be multiple issues with that code.

**Pull request merged**